### PR TITLE
login and forget password views

### LINKED ISSUE #37: Login view
### Description

Develop the login view where users can input their credentials to log into the system.

### Tasks

- Create the UI layout for the login view.
- Integrate the backend functionality to authenticate users.
- Validate user inputs and show appropriate error messages.

### Design
- image

### LINKED ISSUE #38: Forget password view
### Description

Develop the forget password view where users can input their email and send messages over email.

### Tasks

- Create the UI layout for the forget password view.
- Integrate the backend functionality to send an email to the entered email.
- Validate user input and show an appropriate error message.

### Design
- Should be one input and button and an alert message

### PR DESCRIPTION
**Pull request by asmaaadel0**
### Description:
Develop the login and forget passwords views.

### Issue link:
- [Login view issue](https://github.com/codescalersinternships/Flyspray/issues/37)
- [Forget password view issue](https://github.com/codescalersinternships/Flyspray/issues/38)

### Design
- Login vue
  - if user enter invalid email
invalid email

  - if user enter invalid password
invalid pass

  - if input in valid
valid input

- Forget password view

  - if user enter invalid input
image

  - if input is valid
image

### CODE REVIEW

File: frontend/src/components/LogoImage.vue [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<template>
+  <div>
+    <v-img src="../assets/logo.png" alt="logo" class="img"></v-img>
```
Comment (mahmoud-emad) [inline frontend/src/components/LogoImage.vue]: will be better if we renamed this to auth_logo 
We'll have a logo for the project in the future

File: frontend/src/components/auth/InputForm.vue [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<template>
+  <div>
+    <slot></slot>
+  </div>
+</template>
+
+<script lang="ts">
+export default {};
+</script>
+
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/InputForm.vue:10]: Let's merge the image component to this component, and rename it to AuthComponent

File: frontend/src/components/auth/LoginCard.vue [hunk 3]
```diff
@@ -0,0 +1,138 @@
+<template>
+  <div class="login-box">
+    <div class="header-box">
+      <p class="header-text" id="login-card-title">Welcome Back to FLYSPRAY</p>
```
Comment (Mahmoud-Emad) [inline frontend/src/components/auth/LoginCard.vue:4]: Welcome back to FlySpray

File: frontend/src/components/auth/LoginCard.vue [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+              type="password"
+              prepend-inner-icon="mdi-lock"
+              label="Password"
+              v-model="password"
+              required
+              :rules="[validatePassword]"
+            ></v-text-field>
+
+            <div class="forgot-password-container">
+              <router-link to="/forget" class="link"
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/LoginCard.vue]:
```suggestion
              <router-link to="/forget-password" class="link"
```

File: frontend/src/components/auth/LoginCard.vue [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+              block
+              class="mt-2 btn"
+              :disabled="errorPassword || errorEmail"
+              >Sign In</v-btn
+            >
+          </v-form>
+          <hr class="form-separator" />
+
+          <p class="signin-text">
+            Don't have account? please
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/LoginCard.vue]:
```suggestion
            Don't have an account? please
```

File: frontend/src/components/auth/LoginCard.vue [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  methods: {
+    validateEmail(value: string) {
+      if (!value) return "Email is required";
+      const pattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+      return pattern.test(value) || "Invalid email format";
+    },
+    validatePassword(value: string) {
+      if (!value) return "Password is required";
+      return value.length >= 8 || "Password must be at least 8 characters long";
+    },
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/LoginCard.vue]: in the future we'll have huge validations, What do think if we moved these functions to a separate file called `validations.ts` inside the `utils` folder

also the the validation should return two values (bool -> if valid , errorMessage if there is an error)

e.g.

```ts

type ValidationResult = {
  isValid: boolean;
  errorMessage?: string;
};

export function validateEmail(value: string): ValidationResult {
  if (!value || value.trim() === "") {
    return {
      isValid: false,
      errorMessage: "Email is required",
    };
  }

  // Implement email validation logic here
  // Example: Check if the value follows a valid email format
  // You can use regular expressions or any other validation logic

  if (!isValidEmailFormat(value)) {
    return {
      isValid: false,
      errorMessage: "Invalid email format",
    };
  }

  // Add more validation rules as needed

  return {
    isValid: true,
  };
}
```

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]
Fantastic job! I truly appreciate your effort 😄👏. I've gone ahead and added a few comments, so it would be awesome if you could take a look. Keep up the awesome work!

**Comment by Mahmoud-Emad** [comment]
Also, Please let's follow the style form | Font, Color, Icons, and Spaces

### Figma 
- image

### Yours
- image

File: frontend/src/components/auth/LoginCard.vue (hunk 3 above)
Comment (Mahmoud-Emad) [reply frontend/src/components/auth/LoginCard.vue:4]: Don't take this one

File: frontend/src/components/auth/InputForm.vue (hunk 2 above)
Comment (asmaaadel0) [reply frontend/src/components/auth/InputForm.vue:10]: I import image component in login view, and this component in login card, how can I merge them?

File: frontend/src/components/auth/ForgetCard.vue [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            <v-text-field
+              prepend-inner-icon="mdi-account"
+              label="Email"
+              v-model="email"
+              required
+              :rules="[validateEmailRule]"
+            ></v-text-field>
+
+            <v-btn type="submit" block class="mt-2 btn" :disabled="errorEmail"
+              >Continue</v-btn
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/ForgetCard.vue]: Reset password

File: frontend/src/components/auth/ForgetCard.vue [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+            >
+          </v-form>
+          <div class="container">
+            <p class="signin-text">We will send forget password link to you.</p>
```
Comment (mahmoud-emad) [inline frontend/src/components/auth/ForgetCard.vue]:
```suggestion
            <p class="signin-text">We'll send you a link to reset your password</p>
```

File: frontend/src/utils/validations.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
+export type ValidationResult = {
+  isValid: boolean;
+  errorMessage: string;
+};
+
+function isValidEmailFormat(value: string): boolean {
+  const pattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
```
Comment (mahmoud-emad) [inline frontend/src/utils/validations.ts]: Move that pattern at the top of the file and rename it to `R_PASSWORD`

File: frontend/src/utils/validations.ts [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+  return {
+    isValid: true,
+    errorMessage: "",
+  };
+}
+
+function isValidPassword(value: string): boolean {
+  return value.length >= 8;
+}
```
Comment (mahmoud-emad) [inline frontend/src/utils/validations.ts]: I think this is not needed, you can validate the password by accessing the isValid field.

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]

File: frontend/src/utils/validations.ts [hunk 11]
```diff
@@ -0,0 +1,52 @@
+const R_PASSWORD = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+export type ValidationResult = {
+  isValid: boolean;
+  errorMessage: string;
+};
+
+function isValidEmailFormat(value: string): boolean {
```
Comment (mahmoud-emad) [inline frontend/src/utils/validations.ts]: This is not needed

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]

**Review by Mahmoud-Emad — APPROVED** [review]

**Pull request merged**